bindings/py/nupic/bindings/check.py

```diff
@@ -22,39 +22,46 @@
 
 
 def checkImportBindingsInstalled():
-  try:
-    import nupic.bindings
-  except ImportError:
-    return False
-  else:
-    return True
+  """Check that nupic.bindings is installed (can be imported).
+
+  Throws ImportError on failure.
+  """
+  import nupic.bindings
 
 
 
 def checkImportBindingsExtensions():
-  try:
-    import nupic.bindings.math
-    import nupic.bindings.algorithms
-    import nupic.bindings.engine_internal
-  except ImportError:
-    return False
-  else:
-    return True
+  """Check that nupic.bindings extension libraries can be imported.
+
+  Throws ImportError on failure.
+  """
+  import nupic.bindings.math
+  import nupic.bindings.algorithms
+  import nupic.bindings.engine_internal
 
 
 
 def checkMain():
   """
   This script performs two checks. First it tries to import nupic.bindings
   to check that it is correctly installed. Then it tries to import the C
-  extensions under nupic.bindings. Appropriate user-friendly status messages 
+  extensions under nupic.bindings. Appropriate user-friendly status messages
   are printed depend on the outcome.
   """
-  if checkImportBindingsInstalled():
-    if checkImportBindingsExtensions():
-      print "Nupic bindings could be successfully imported."
-    else:
-      print ("Could not import C extensions for nupic.bindings. " +
-        "Make sure that the package was properly installed.")
-  else:
-    print "Could not import nupic.bindings. It must be installed before use."
+  try:
+    checkImportBindingsInstalled()
+  except ImportError as e:
+    print ("Could not import nupic.bindings. It must be installed before use. "
+           "Error message:")
+    print e.message
+    return
+
+  try:
+    checkImportBindingsExtensions()
+  except ImportError as e:
+    print ("Could not import C extensions for nupic.bindings. Make sure that "
```

Comment:
@scottpurdy: by the way, would it be clear from these messages which of the three extensions failed to load? A traceback would make it explicit.